src/src/rx_main.cpp

```diff
@@ -1012,51 +1012,89 @@ void UpdateModelMatch(uint8_t model)
         deferExecution(500, []() {
             setWifiUpdateMode();
         });
 #endif
-    }
-    else if (MspData[0] == MSP_ELRS_SET_RX_LOAN_MODE)
-    {
+        break;
+    case MSP_ELRS_SET_RX_LOAN_MODE: //0x0F
         loanBindTimeout = LOAN_BIND_TIMEOUT_MSP;
         InLoanBindingMode = true;
-    }
-    else if (OPT_HAS_VTX_SPI && MspData[7] == MSP_SET_VTX_CONFIG)
-    {
-        vtxSPIBandChannelIdx = MspData[8];
-        if (MspData[6] >= 4) // If packet has 4 bytes it also contains power idx and pitmode.
-        {
-            vtxSPIPowerIdx = MspData[10];
-            vtxSPIPitmode = MspData[11];
-        }
-        devicesTriggerEvent();
-    }
-    else
-    {
-        crsf_ext_header_t *receivedHeader = (crsf_ext_header_t *) MspData;
-
-        // No MSP data to the FC if no model match
-        if (connectionHasModelMatch && (receivedHeader->dest_addr == CRSF_ADDRESS_BROADCAST || receivedHeader->dest_addr == CRSF_ADDRESS_FLIGHT_CONTROLLER))
-        {
-            crsf.sendMSPFrameToFC(MspData);
-        }
-
-        if ((receivedHeader->dest_addr == CRSF_ADDRESS_BROADCAST || receivedHeader->dest_addr == CRSF_ADDRESS_CRSF_RECEIVER))
-        {
-            crsf.ParameterUpdateData[0] = MspData[CRSF_TELEMETRY_TYPE_INDEX];
-            crsf.ParameterUpdateData[1] = MspData[CRSF_TELEMETRY_FIELD_ID_INDEX];
-            crsf.ParameterUpdateData[2] = MspData[CRSF_TELEMETRY_FIELD_CHUNK_INDEX];
-            luaParamUpdateReq();
+        break;
+    default:
+        switch(receivedHeader->type) {
+        case CRSF_FRAMETYPE_MSP_WRITE: //encapsulated MSP payload
+            if (MspData[7] == MSP_SET_RX_CONFIG && MspData[8] == MSP_ELRS_MODEL_ID)
+            {
+                UpdateModelMatch(MspData[9]);
+            }
+            else if (OPT_HAS_VTX_SPI && MspData[7] == MSP_SET_VTX_CONFIG)
+            {
+                vtxSPIBandChannelIdx = MspData[8];
+                if (MspData[6] >= 4) // If packet has 4 bytes it also contains power idx and pitmode.
+                {
+                    vtxSPIPowerIdx = MspData[10];
+                    vtxSPIPitmode = MspData[11];
+                }
+                devicesTriggerEvent();
+            }
+            break;
+        case CRSF_FRAMETYPE_PARAMETER_SETTINGS_ENTRY:
+            if ((receivedHeader->dest_addr == CRSF_ADDRESS_BROADCAST || receivedHeader->dest_addr == CRSF_ADDRESS_CRSF_RECEIVER))
+            {
+                crsf.ParameterUpdateData[0] = MspData[CRSF_TELEMETRY_TYPE_INDEX];
+                crsf.ParameterUpdateData[1] = MspData[CRSF_TELEMETRY_FIELD_ID_INDEX];
+                crsf.ParameterUpdateData[2] = MspData[CRSF_TELEMETRY_FIELD_CHUNK_INDEX];
+                luaParamUpdateReq();
+            }
+            break;
+        default:
+            // No MSP data to the FC if no model match
+            if (connectionHasModelMatch && (receivedHeader->dest_addr == CRSF_ADDRESS_BROADCAST || receivedHeader->dest_addr == CRSF_ADDRESS_FLIGHT_CONTROLLER))
```

Comment:
The default case should be moved after the switch block. The way it's implemented now would prevent sending CRSF_FRAMETYPE_MSP_WRITE or CRSF_FRAMETYPE_PARAMETER_SETTINGS_ENTRY to the FC (I don't really know a usecase for that but people try random stuff and this would prevent that from working).

Reply:
Correct, well found. I'll move the default case out.

I propose to also remove the default case on the first byte and replace it with:

```
case CRSF_ADDRESS_CRSF_TRANSMITTER:
case CRSF_SYNC_BYTE:
```

Such that only CRSF packets are processed by the inner switch. What do you think?

Reply:
I left the default case in. As far as I can tell actually the following first byte values are possible for CRSF packages:
```
case CRSF_ADDRESS_CRSF_TRANSMITTER: // from CRSF::handleUARTin()
case CRSF_SYNC_BYTE:  // from CRSF::handleUARTin()
case CRSF_ADDRESS_BROADCAST: // from CRSF::AddMspMessage(mspPacket_t* packet)
```